### PR TITLE
Svc model

### PR DESCRIPTION
**Pull request by sneaky-hobbitses**
Working Support Vector Classifier. 
Instantiate with `model = Model()`
then `model.predict(job description or list of skills)`

### CODE REVIEW

File: data_scientist_skills/skill_extraction.py [hunk 1]
```diff
@@ -10,9 +10,14 @@ def remove_more_stopwords(text):
     clean_text.replace(',', '')
     return clean_text
 
-def tdidf(X, min_df=0.1, max_df=0.6, results_df=False):
+def tdidf(X, min_df=0.1, max_df=0.6, results_df=False, transform=False):
     """Performs TD-IDF on pd Series. Results_df=True to return table of results. """
```
Comment (RemoteSalad) [inline data_scientist_skills/skill_extraction.py]: The shortened 'df' is working for two different meanings here. If not expanding the tdidf method parameters(e.g. min_doc_frequency), can you toss the explanation in the doc string?

Comment (RemoteSalad) [reply data_scientist_skills/skill_extraction.py]: Also can the doc string include your intentions for the transform parameter?

File: data_scientist_skills/skill_extraction.py [hunk 2]
```diff
@@ -10,9 +10,14 @@ def remove_more_stopwords(text):
...
     tfidf = TfidfVectorizer(ngram_range=(1,2), min_df=min_df, max_df=max_df)
+
+    if transform:
+        # tfidf.fit(X)
```
Comment (RemoteSalad) [inline data_scientist_skills/skill_extraction.py]: So if transform is set to True the function exits and gives the initialized but unfit vectorizer? I trust that it's useful but I don't follow this option

Comment (sneaky-hobbitses) [reply data_scientist_skills/skill_extraction.py]: Yeah for some reason I couldn't fit within this function, and then use it to transform the input outside of this function. So simply getting the same model for use on the input

Comment (RemoteSalad) [reply data_scientist_skills/skill_extraction.py]: I see now that it's used that way in the init_vectorizer

File: data_scientist_skills/skill_extraction.py (hunk 1 above)
Comment (sneaky-hobbitses) [reply data_scientist_skills/skill_extraction.py]: Updated docstring

File: data_scientist_skills/model.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    # X and y
+    X = clean_df['cleaned_description'] + df['cleaned_title']
+
+    # Returns X for fitting vectorizer, to transform input
+    if return_x:
+        return X
+
+    X_svc = tdidf(X, results_df=True)
+    y_svc = label_encoder(processed_df, 'Category')['lenc_Category']
+    return X_svc, y_svc
```
Comment (RemoteSalad) [inline data_scientist_skills/model.py]: Why are X and y tagged with svc? The term has overlap with the Support Vector Classification model and I don't think the meaning is the same.

Is the phrasing of X_svc and y_svc just to telegraph that they are ready to be put into the SVC model?

Comment (sneaky-hobbitses) [reply data_scientist_skills/model.py]: Yep correct, just saying that they're good to go for the fitting the model

Comment (sneaky-hobbitses) [reply data_scientist_skills/model.py]: Yep correct, just saying that they're good to go for the fitting the model

**Comment by honnsey** [comment]
all good, i have no further comments!

**Pull request merged**